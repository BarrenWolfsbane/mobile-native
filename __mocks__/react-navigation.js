export default {

};

export const withNavigation = jest.fn();

export const Header = {
  HEIGHT:80
}

export const StackActions = {
  reset: jest.fn()
}
export const NavigationActions = {
  navigate: jest.fn(),
  setParams: jest.fn(),
}