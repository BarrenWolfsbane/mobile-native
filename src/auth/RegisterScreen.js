import React, { Component } from 'react';

import { View } from 'react-native';

import RegisterForm from './RegisterForm';
import { CommonStyle } from '../styles/Common';
import logService from '../common/services/log.service';

/**
 * Register screen
 */
export default class RegisterScreen extends Component {
  /**
   * Disable navigation bar
   */
  static navigationOptions = {
    header: null,
  };

  /**
   * Render
   */
  render() {
    return (
      <View style={[CommonStyle.flexContainer]}>
        <RegisterForm onRegister={this.onRegister} onBack={this.onPressBack} />
      </View>
    );
  }

  /**
   * On press back
   */
  onPressBack = () => {
    this.props.navigation.navigate('Login');
  };

  /**
   * On register
   */
  onRegister = guid => {
    logService.info('[Register] new user registered ' + guid);
  };
}
