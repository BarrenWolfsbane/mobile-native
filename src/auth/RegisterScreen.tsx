import React from 'react';
import { View, Platform, Text } from 'react-native';

import { StackNavigationProp } from '@react-navigation/stack';
import { observer, useLocalStore } from 'mobx-react';
import Icon from 'react-native-vector-icons/Ionicons';

import InputContainer from '../common/components/InputContainer';
import { AuthStackParamList } from '../navigation/NavigationTypes';
import { styles, shadowOpt, icon } from './styles';
import i18n from '../common/services/i18n.service';
import ThemedStyles from '../styles/ThemedStyles';
import BoxShadow from '../common/components/BoxShadow';
import { SafeAreaView } from 'react-native-safe-area-context';
import Button from '../common/components/Button';
import PasswordValidator from '../common/components/PasswordValidator';

export type WalletScreenNavigationProp = StackNavigationProp<
  AuthStackParamList,
  'Register'
>;

type PropsType = {
  navigation: WalletScreenNavigationProp;
};

export default observer(function RegisterScreen(props: PropsType) {
  const store = useLocalStore(() => ({
    error: {},
    password: '',
    passwordFocused: false,
    username: '',
    email: '',
    termsAccepted: false,
    exclusivePromotions: false,
    hidePassword: true,
    inProgress: false,

    setPassword(value: string) {
      store.password = value;
    },
    setUsername(value: string) {
      store.username = value;
    },
    setEmail(value: string) {
      store.email = value;
    },
    toggleTerms() {
      store.termsAccepted = !store.termsAccepted;
    },
    toggleHidePassword() {
      store.hidePassword = !store.hidePassword;
    },
    togglePromotions() {
      store.exclusivePromotions = !store.exclusivePromotions;
    },
  }));

  const theme = ThemedStyles.style;

  const inputs = (
    <SafeAreaView style={styles.shadow}>
      <View style={[theme.rowJustifyStart, theme.paddingVertical3x]}>
        <Text
          style={[
            theme.titleText,
            theme.textCenter,
            theme.colorWhite,
            theme.paddingVertical3x,
            theme.positionAbsolute,
          ]}>
          {i18n.t('auth.createChannel')}
        </Text>
        <Icon
          size={34}
          name="ios-chevron-back"
          style={[theme.colorWhite, theme.padding1x]}
          onPress={props.navigation.goBack}
        />
      </View>
      <InputContainer
        containerStyle={styles.inputBackground}
        placeholder={i18n.t('auth.username')}
        onChangeText={store.setUsername}
        value={store.username}
        testID="usernameInput"
        noBottomBorder
      />
      <InputContainer
        containerStyle={styles.inputBackground}
        placeholder={i18n.t('auth.email')}
        onChangeText={store.setEmail}
        value={store.email}
        testID="email"
        noBottomBorder
      />
      <PasswordValidator password={store.password} />
      <View>
        <InputContainer
          containerStyle={styles.inputBackground}
          placeholder={i18n.t('auth.password')}
          secureTextEntry={store.hidePassword}
          onChangeText={store.setPassword}
          value={store.password}
          testID="userPasswordInput"
        />
        <Icon
          name={store.hidePassword ? 'md-eye' : 'md-eye-off'}
          size={25}
          onPress={store.toggleHidePassword}
          style={[theme.inputIcon, icon]}
        />
      </View>
      <View style={theme.padding4x}>
        <Button
          onPress={store.onRegisterPress}
          text={i18n.t('auth.createChannel')}
          containerStyle={[
            theme.transparentButton,
            theme.paddingVertical3x,
            theme.fullWidth,
            theme.marginTop6x,
            styles.lightButton,
          ]}
          textStyle={theme.buttonText}
          loading={store.inProgress}
          disabled={store.inProgress}
          testID="loginButton"
        />
      </View>
    </SafeAreaView>
  );

  const inputsWithShadow = Platform.select({
    ios: inputs,
    android: <BoxShadow setting={shadowOpt}>{inputs}</BoxShadow>, // Android fallback for shadows
  });

  return <View style={theme.flexContainer}>{inputsWithShadow}</View>;
});
