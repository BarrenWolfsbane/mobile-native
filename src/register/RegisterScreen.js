import React, {
  Component
} from 'react';

import { NavigationActions } from 'react-navigation';
import FastImage from 'react-native-fast-image';

import {
  Text,
  StyleSheet,
  KeyboardAvoidingView,
  ScrollView,
  View,
  Image
} from 'react-native';

import RegisterForm from './RegisterForm';
import { CommonStyle } from '../styles/Common';
import { ComponentsStyle } from '../styles/Components';

export default class Register extends Component {

  /**
   * Disable navigation bar
   */
  static navigationOptions = {
    header: null
  }

  render() {
    return (
      <View style={CommonStyle.flexContainer}>
        <FastImage
          resizeMode={FastImage.resizeMode.cover}
          style={ComponentsStyle.backgroundImage}
          source={require('../assets/bg-2.jpg')}
        />
        <View style={[CommonStyle.flexContainerCenter, CommonStyle.padding2x]}>
          <FastImage
            resizeMode={FastImage.resizeMode.cover}
            style={ComponentsStyle.logo}
            source={require('../assets/logos/medium-white.png')}
          />
          <RegisterForm
            onRegister={() => this.onRegister()}
            onBack={() => this.onPressBack()}
          />

        </View>
    </View>
    );
  }

  onPressBack() {
    this._navigate('Login');
  }

  onRegister() {
    this._navigate('Tabs');
  }

  /**
   * Navigate to screen
   * @param {string} destination
   */
  _navigate(destination) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [
        NavigationActions.navigate({ routeName: destination })
      ]
    })

    this.props.navigation.dispatch(resetAction);
  }
}

const styles = StyleSheet.create({
  stretch: {
    width: 200,
    height: 80,
    marginLeft: 20,
    marginTop: 0
  },
  background: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    justifyContent: 'center',
    alignItems: 'center',
  },
  viewWrapper: {
    flex: 1,
    backgroundColor: 'transparent',
    justifyContent: 'center',
  },
  keyboardAvoidingView: {backgroundColor: 'transparent'}
});