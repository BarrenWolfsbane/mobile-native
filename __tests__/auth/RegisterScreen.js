import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import RegisterScreen from '../../src/auth/RegisterScreen';
import { NavigationActions, StackActions } from 'react-navigation';

jest.mock('../../src/auth/RegisterForm', () => 'RegisterForm');
jest.mock('../../src/common/components/VideoBackground', () => 'VideoBackground');

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('RegisterScreen component', () => {
  it('should renders correctly', () => {
    const loginScreen = renderer.create(
      <RegisterScreen />
    ).toJSON();
    expect(loginScreen).toMatchSnapshot();
  });

  it('should shows register form component', async () => {
    const wrapper = shallow(
      <RegisterScreen />
    );

    // find register form
    const registerForms = wrapper.find('RegisterForm');

    // should contain 1 login form
    expect(registerForms.length).toBe(1);
  });

  it('should navigate on RegisterForm events', async () => {

    const navigation = {
      dispatch: jest.fn()
    };

    const wrapper = shallow(
      <RegisterScreen navigation={navigation}/>
    );

    StackActions.reset.mockReturnValue(1)

    // find register form
    const registerForms = wrapper.find('RegisterForm');

    // should contain 1 login form
    expect(registerForms.length).toBe(1);

    // simulate onBack event
    registerForms.at(0).props().onBack();

    // should call navigate
    expect(StackActions.reset).toBeCalledWith({"actions": [undefined], "index": 0});

    // with a navigation action with route to Login
    expect(NavigationActions.navigate).toBeCalledWith({"routeName": "Login"});
  });
});
