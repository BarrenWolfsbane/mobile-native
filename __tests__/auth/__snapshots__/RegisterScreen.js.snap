// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen component should renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "padding": 10,
      },
    ]
  }
>
  <VideoBackground />
  <View
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 0.3,
          },
        ],
      }
    }
  >
    <View>
      <RegisterForm
        onBack={[Function]}
        onRegister={[Function]}
      />
    </View>
  </View>
</View>
`;
